tests/integration/conftest.py

```
@@ -0,0 +1,250 @@
+        except Exception as e:
+            last_err = e
+        time.sleep(HEALTH_POLL_INTERVAL_S)
+    raise RuntimeError(
+        f"orchestrator at {url}/health never became healthy in {timeout}s "
+        f"(last error: {last_err})"
+    )
+
+
+def _compose(cmd: list[str], env: dict | None = None) -> subprocess.CompletedProcess:
+    full = ["docker", "compose", "-f", str(COMPOSE_FILE), *cmd]
+    return subprocess.run(full, capture_output=True, text=True, env={**os.environ, **(env or {})})
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)